### PR TITLE
Minor updates to address comments on #2

### PR DESCRIPTION
**Pull request by scasplte2**
## Purpose
Address minor feedback in review of #2

## Approach
- Introduce `KnownReference` to distinguish spendable and non-spendable values that may be queried in state
- minor typos
- added Signable bytes for new types
- removed deprecated code
- formatted code
- added some comments to files where appropriate
- move Identifier names to be hard-wired instead of optional parameters
- moved `blobs` default assignmant in Box.Value
- renamed BodyDatum to RootDatum (may need to add Body datum back at some point)
- refactored Proposer and Prover to return the specific model corresponding to each instance
- removed `F` wrapper on `ParsableDataInterface` as it was unused

### CODE REVIEW

File: src/main/scala/co/topl/node/KnownReference.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    indices:   List[Int],
+    reference: Reference[Events.SpentTransactionOutput]
+  ) extends KnownReference[Events.SpentTransactionOutput]
+
+  case class Predicate64(
+    network:   Int,
+    ledger:    Int,
+    indices:   List[Int],
+    reference: Reference[Events.SpentTransactionOutput]
+  ) extends KnownReference[Events.SpentTransactionOutput]
```
Comment (mgrand-topl) [inline src/main/scala/co/topl/node/KnownReference.scala:24]: Why distinguish between `Predicate32` and `Predicate64` when they are defined to contain _exactly_ the same thing?

**Review by mgrand-topl — APPROVED** [review]
It is simpler to follow conversations if you put changes in response to feedback in the same pull request. This will not make it difficult for reviewers, because the comparison has an option to see only commits since the reviewer's last review.

**Review by DiademShoukralla — APPROVED** [review]
Looks good to me

File: src/main/scala/co/topl/node/KnownReference.scala (hunk 1 above)
Comment (scasplte2) [reply src/main/scala/co/topl/node/KnownReference.scala:24]: good point, we can simplify this

**Pull request merged**